### ISSUE TITLE
Bad `identifier` attribute on vehicle.lastStation and vehicle.nextStation

### ISSUE THREAD

**Issue by elliottwilliams** [propershark/shark#17]
After pulling the latest shark (https://github.com/propershark/shark/commit/2d879e8efbf1f5b883aaef5481d42061e17f5b51) I started getting Vehicle decoder errors. Some investigation revealed objects like this vehicle (note the last_station and next_station identifiers):

``` json
{
  "saturation": 0,
  "code": null,
  "capacity": 60,
  "name": "4008",
  "route": {
    "color": "1E90FF",
    "identifier": "routes.1A",
    "name": "Market Square",
    "short_name": "1A"
  },
  "associated_objects": {},
  "longitude": null,
  "heading": 270.1400146484375,
  "last_station": {
    "identifier": "stations.BUS766W",
    "name": "Sickle Ct & McCarty Ln (NW Corner) - BUS766W"
  },
  "latitude": null,
  "schedule_delta": null,
  "onboard": 0,
  "speed": 12.7324074,
  "next_station": {
    "identifier": {
      "identifier": "stations.BUS766W",
      "name": "Sickle Ct & McCarty Ln (NW Corner) - BUS766W"
    },
    "name": "McCarty Ln & Emerald Pines Ct (NE Corner)- BUS774N"
  }
}
```

Perhaps something like `next_station.identifier = last_station` is happening by accident?

**Comment by faultyserver** [comment]
I think this might be fixed on the `object_embed` branch (to be merged before too long). The code looked like this:

``` ruby
vehicle.next_station = begin
  station = @storage.find(vehicle.next_station)
  { identifier: vehicle.next_station, name: station&.name }
end
```

which was replacing the actual `last_station` attribute on `vehicle`, rather than on the hash representation of Vehicle (another reason for #14 and the current state of #11). This code means that if `next_station` doesn't get modified in a refresh cycle, you'll end up with the nested hash that you're seeing.

In the example you gave, it looks like `last_station` is fine, presumably because it _did_ get modified in the previous cycle.

This will definitely be fixed before `object_embed` gets merged. I'll tag this issue in the associated PR.

**Comment by faultyserver** [comment]
But, it seems that I didn't address the main question you asked: it looks like `last_station == next_station`.

This could be because `last_station` is sourced from DoubleMap, while `next_station` is sourced from TripSpark, and their data sets could be out of sync by a few seconds.

If the problem persists passed 5 or so update cycles, that's another issue, but my guess is just synchronization between the sources. It shouldn't cause any issues, since the two attributes are never used in combination (at least server-side).

I trust DoubleMap's data more anyway; since they're the ones that provide the GPS system for the buses, they're likely to have more up-to-date information, if only by a few seconds.

**Comment by faultyserver** [comment]
Closing, because this seems to be fixed via #18.

Note that some data may still have this incorrect structure if it is stale (e.g., the vehicle hasn't been active in the past few days), but it should all be correct eventually, once everything has had at least one update cycle.

Alternatively, we can restart crossbar to clear the cache.
